### PR TITLE
Math with protocols - Compile errors

### PR DESCRIPTION
**Pull request by BigSpaces**
Hello @brooklinjazz,

I would like help diagnosing these compile errors. Why, when using the same code in a new cell, the code runs without trouble? Am I to diagnose this as a Livebook bug, or am I actually doing something wrong somewhere? 

Thanks for your support.

image

### CODE REVIEW

File: exercises/math_with_protocols.livemd [hunk 1]
```diff
@@ -96,62 +96,88 @@ Define implementations for the `Math` protocol as documented below.
 
 ```elixir
 defprotocol Math do
-  @doc """
-  Add two integers, lists, or strings.
+  def add(value1, value2)
+end
+
+@doc """
```
Comment (BrooklinJazz) [inline exercises/math_with_protocols.livemd:102]: You have an @doc outside of a module.

While this is the same color as a comment, it's not a comment. It's the same as defining any module attribute but the Elixir highlighter is smart enough to make it grey when it's an `@doc` module attribute.

File: exercises/math_with_protocols.livemd [hunk 2]
```diff
@@ -96,62 +96,88 @@ Define implementations for the `Math` protocol as documented below.
...
+
+defimpl Math, for: Integer do
+  def add(value1, value2) do
+    value1 + value2
+  end
+end
+
+# end
+
+@doc """
```
Comment (BrooklinJazz) [inline exercises/math_with_protocols.livemd:137]: Same issue with @doc

**Review by BrooklinJazz — CHANGES_REQUESTED** [review]
You have some documentation that you removed from the module; that's what's causing your compile error.

For issues like this, I recommend the "Shine the light" method of debugging.

Remove portions of your code and narrow down on where exactly the issue is.

**Comment by BigSpaces** [comment]
Thank you @BrooklinJazz . This helps with understanding of documentation and testing

**Pull request closed without merging**